pages/behaviors/[behaviorId].tsx

```diff
@@ -4,47 +4,56 @@ import Navbar from "../../components/Navbar";
 import Link from "next/link";
 import { Button } from "@material-ui/core";
 import CheckUser from "../../auth0CheckUser";
+import { useState, useEffect } from "react";
+import { useRouter } from 'next/router';
 
-function BehaviorPage({ behaviorData: { data } }) {
+function BehaviorPage() {
     // Verifies if user has the correct permissions
     const {allowed, role} = CheckUser(["Admin", "BCBA"])
     if(!allowed) return(<div>Redirecting...</div>);
+
+    const router = useRouter();
+    const { behaviorId } = router.query;
+
+    const [behaviorData, setBehaviorData] = useState([])

```

Comment:
Rather than having the type of the state be any[] here, I would have it be specifically the Behavior type used elsewhere in the code. This will also get rid of the typescript errors that you are getting below. 

At the top of the file, I would import the type like so:
`import { Behavior } from '@prisma/client';`

And then when declaring my state, I would do it like so:
const [behaviorData, setBehaviorData] = useState<Behavior | null >(null)

Reply:
If I initialize the useState with null, I run into a "TypeError: Cannot read properties of null (reading 'name')" on the line <Navbar pageTitle={behaviorData.name} role={role}>, but if initialized with either an empty [] or {}, it works fine although the vscode says "Property 'name' does not exist on type '(GetResult<{ id: number; name: string; description: string; datatype: Datatype; trialsPerEntry: number; entries: string[]; tags: string[]; }, unknown> & {}) | []'.
  Property 'name' does not exist on type '[]'.ts(2339)"

Reply:
Does this mean I need an additional data processing for the following data object? data: Behavior = await response.json();
setBehaviorData(data);

Reply:
So what I think is happening is that the data is null when the component is first rendered because we are still fetching data from the api. This is normal in web applications, that's why we have stuff like loading spinners.

I would initialize useState with null, and then conditionally change what you are rendering based on whether your state is null. If the data is null, render a loading indicator, if not, render your data.


Here is a short example to showcase what I am talking about:
![image](https://github.com/UTDallasEPICS/ATC-Patient-Data/assets/63270519/df1ce1e7-02d5-4e30-b436-8337e706433f)

Reply:
It might also help to have optional chaining here. 

For example, instead of this `<h1>Behavior Name: {behaviorData.name}</h1>`, 
do this: `<h1>Behavior Name: {behaviorData?.name}</h1>`